kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -218,16 +218,23 @@ prove
     -> m ()
 prove cindex = do
     claim' <- getClaimByIndex . unClaimIndex $ cindex
-    maybe printNotFound initProof claim'
+    maybe printNotFound (initOrSwitchToProof cindex) claim'
```

Comment:
As said, I would just call `switchToClaim` here which switches to requested claim, and initializes/updates all related fields.